### DISCUSSION TITLE
How to Update BabyBuddy

### DISCUSSION THREAD

**Discussion by five2seven** [babybuddy/babybuddy#593] [Support]
I have a manual install on a VPS. Just realized it's super out of date and there are new features and I'd like to try out. How can I update my BabyBuddy instance without overwriting my database?

**Comment by cdubz** [comment]
When you say you did a manual deploy do you mean that you followed the manual deploy [example deployment](https://docs.baby-buddy.net/setup/deployment/#example-deployment)? Updating from that does generally require a few little adjustments here and there but it can depend on how closely you followed the example.

Here is a draft of instructions for updating from following the example. Note I have not tested this!

0. Back up your database

       sudo cp /var/www/babybuddy/data/db.sqlite3 /var/www/babybuddy/data/db.sqlite3.bak

1. Navigate to the directory where Baby Buddy's code is cloned from git.

       cd /var/www/babybuddy/public

2. Fetch and pull the latest updates.

       git fetch && git pull

3. Update Python dependencies and enter the pipenv shell.

       export PIPENV_VENV_IN_PROJECT=1
       pipenv install
       pipenv shell

4. Make the database writable by your user (there are safer ways to do this and this step is a PITA, but this is the fastest).

       sudo chmod 666 /var/www/babybuddy/data/db.sqlite3
       sudo chmod 777 /var/www/babybuddy/data

5. Run any database migrations.

       export DJANGO_SETTINGS_MODULE=babybuddy.settings.production
       python manage.py migrate

6. Reset minimal permissions on database file.

       sudo chmod 640 /var/www/babybuddy/data/db.sqlite3
       sudo chmod 750 /var/www/babybuddy/data

7. Maybe restart uwsgi and nginx?

       sudo service uwsgi restart
       sudo service nginx restart

After that you should be good. If anything went wrong you can checkout whatever revision you were on previously and restore from the backup in step 0.

This is definitely a process that should be improved 😄

**Reply by five2seven** [reply]
This process worked for me last time (thank you!) but this time I'm getting a 500 error from BB after logging in. 
![Screenshot_1](https://github.com/babybuddy/babybuddy/assets/44069152/6e7c7d72-b862-4000-96bf-88b9e71ec6a8)

**Reply by five2seven** [reply]
Solved! Adding in case it helps someone else: I had to run the following commands one more time. 

`python manage.py migrate` 
then 
`python manage.py createcachetable`
then 
`service uwsgi restart`

**Comment by fs1** [comment]
Getting here from #65.
These are the commands I used for updating (to 1.15.0):
```
# get the newest babybuddy version
cd /download/location/
curl -LO https://github.com/babybuddy/babybuddy/archive/refs/tags/v1.15.0.tar.gz

# backup database (postgres in my case)
pg_dump babybuddy > /path/to/backup.sql

cd /opt/babybuddy
mv public/ /path/to/backup/
tar xzf /download/location/babybuddy-1.15.0.tar.gz
mv babybuddy-1.15.0/ public
cp /path/to/backup/public/babybuddy/settings/production.py public/babybuddy/settings/
pipenv install --three
pipenv shell
export DJANGO_SETTINGS_MODULE=babybuddy.settings.production
python manage.py makemigrations
python manage.py migrate
exit
systemctl restart babybuddy.service
```

- So my installation is under /opt/babybuddy, but if you replace the path with your location, this should work fine.
- babybuddy.service is a babybuddy gunicorn daemon.
- For professional use, I recommand denying access between the database update and restarting the babybuddy service. But since I do not consider missing babybuddy-entries critical (like most people will do I guess), I omitted shutting down the webserver.

**Comment by cdubz** [comment]
FYI I added #898 after doing an update for my own manual deployment (and forgetting about this thread). I'll try to get this in the actual docs and maybe see if I can script it...
